packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.types.ts

```diff
@@ -451,10 +451,14 @@ export interface IContextualMenuItem {
    * (like an anchor or a button) or have the `data-is-focusable` property set to true.
    *
    * The function receives a function that can be called to dismiss the menu as a second argument.
-   *  This can be used to make sure that a custom menu item click dismisses the menu.
+   * This can be used to make sure that a custom menu item click dismisses the menu.
+   *
+   * posInSet and setSize are passed to the function for cases where the rendered item needs to be
+   * able to set these aria attributes (like in the command bar)
+   *
    * @defaultvalue undefined
    */
-  onRender?: (item: any, dismissMenu: (ev?: any, dismissAll?: boolean) => void) => React.ReactNode;
+  onRender?: (item: any, dismissMenu: (ev?: any, dismissAll?: boolean) => void, posInSet?: number, setSize?: number) => React.ReactNode;
```

Comment:
Be careful with how this interacts with customlistlength. If someone has a list item which actually renders multiple items, the posInSet might not be accurate.

Reply:
@joschect are you asking for a change here? Or documentation update? How can we help avoid this problem you're describing?

Reply:
There are two number arguments which are optional, which reduces arg safety quite a bit. The calling function has no idea if a caller provided only posInSet and not setSize or omitted posInSet and is specifying setSize.

Reply:
(In addition to Jon's concerns)